official/transformer/v2/transformer_main.py

```diff
@@ -107,13 +113,22 @@ def train(self):
     """Trains the model."""
     params, flags_obj, is_train = self.params, self.flags_obj, True
     _ensure_dir(flags_obj.model_dir)
-    model = transformer.create_model(params, is_train)
-    opt = self._create_optimizer()
+    if self.distribution_strategy:
+      with self.distribution_strategy.scope():
+        model = transformer.create_model(params, is_train)
+        opt = self._create_optimizer()
+        model.compile(opt)
+    else:
+      model = transformer.create_model(params, is_train)
+      opt = self._create_optimizer()
+      model.compile(opt)
 
-    model.compile(opt, target_tensors=[])
     model.summary()
 
-    map_data_fn = data_pipeline.map_data_for_transformer_fn
+    if self.distribution_strategy:
```

Comment:
Might be nice to have a comment as to why you do this for dist_strat.  This is kind of a learning model and nice to point people to it and they can understand the why.

Reply:
added a comment. I tried hard to write the input pipeline in a way that works in both cases but failed so far. A proper fix may need a change in keras i think. this should not be the end state..